components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Stack/Extensions/NodeIdEx.cs

```diff
@@ -27,12 +27,15 @@ public static class NodeIdEx {
             if (NodeId.IsNull(nodeId)) {
                 return ExpandedNodeId.Null;
             }
-            if (nodeId.NamespaceIndex > 0 && namespaces == null) {
-                throw new ArgumentNullException(nameof(namespaces));
-            }
-            return new ExpandedNodeId(nodeId.Identifier, nodeId.NamespaceIndex,
-                nodeId.NamespaceIndex > 0 ? namespaces.GetString(nodeId.NamespaceIndex) :
-                    null, 0);
+
+            ushort namespaceIndex = nodeId.NamespaceIndex;
+            if (namespaceIndex > 0 && namespaces == null) {
+                namespaceIndex = 0;
+            }
+            return new ExpandedNodeId(nodeId.Identifier, namespaceIndex,
+                namespaceIndex > 0
+                ? namespaces.GetString(namespaceIndex)
+                : null, 0);
         }
```

Comment:
We need to revert this, the namespace 0 has a special meaning, being reserved for the OPC Foundation spec nodes. A namespace name must be provided  if the namespace index is != 0. if the table is not provided is a fatal error and we cannot continue, therefore the exception is required here. Changing all namespaces indexes to 0 in case the table is not present would mess up everything